script.js

```diff
@@ -0,0 +1,618 @@
+                    break;
+                case  'link':
+                    if (!adItem.link || typeof adItem.link !== 'string') {
+                        return false;
+                    }
+                    break;
+                case 'vendor':
+                    if (!adItem.vendor || typeof adItem.vendor !== 'string' || adItem.vendor.length < 1){
+                        return false;
+                    }
+                    break;
+                case 'photoLink':
+                    if (adItem.photoLink){
+                        if (typeof adItem.photoLink !== 'string'){
+                            return false;
+                        }
+                    }
+                    break;
+                case 'hashTags':
+                    if (!adItem.hashTags || !adItem.hashTags.every(tag => typeof tag === 'string')){
+                        return false;
+                    }
+                    break;
+                case 'discount':
+                    if (!adItem.discount || typeof adItem.discount !== 'string'){
+                        return false;
+                    }
+                    break;
+                case 'validUntil':
+                    if (!adItem.validUntil || Object.prototype.toString.call(adItem.validUntil) !== '[object Date]') {
+                        return false;
+                    }
+                    break;
+                case 'rating':
+                    if (adItem.rating) {
+                        if (typeof adItem.rating !== 'number' || (adItem.rating < 0 && adItem.rating > 5)) {
+                            return false;
+                        }
+                    }
+                    break;
+                case 'reviews':
+                    break;
+                default:
+                    return false;
+            }
+            return true;
+        })
+    }
+
+    function addAd(adItem){
+        if (validateAd(adItem)){
+            adList.push(adItem);
+            return true;
+        }
+        return false;
+    }
+
+    function editAd(id, adItem){
+        for (var param in adItem){
+            if (param === 'id' || param === 'createdAt' || param === 'vendor'){
+                console.log('You can\'t change id, vendor or created time');
+                return false;
+            }
+        }
+        if(!validateAd(adItem, Object.keys(adItem))){
+            return false;
+        }
+
+        var editingAd = getAd(id);
+
+        for (var param in adItem){
+            editingAd[param] = adItem[param];
+        }
+
+        return true;
+    }
+
+    function removeAd(id){
+        if (typeof id === 'string'){
+            var index = adList.findIndex(adItem => adItem.id === id);
+
+            if (index !== -1){
+                adList.splice(index, 1);
+                return true;
+            }
+
+            if (index === -1){
+             console.log('Incorrect id!')
+            }
+        }
+        return false;
```

Comment:
Just cosmetic:
```
if (typeof id !== 'string {
    return false;
}

var index = adList.findIndex(adItem => adItem.id === id);
if (index === -1){
      console.log('Incorrect id!');
      return false;
}

adList.splice(index, 1);
return true;
```
In case you use early return it's much more clear what are the conditions of successful function execution, so your code becomes more readable.